src/net/inv.rs

```diff
@@ -0,0 +1,1509 @@
+
+    pub fn set_sync_peers(&mut self, mut peers: HashSet<NeighborKey>) -> () {
+        self.sync_peers.clear();
+        for nk in peers.drain() {
+            self.sync_peers.insert(nk);
+        }
+    }
+
+    /// Can we rely on the inventory from this peer? i.e. is the peer in sync with our view of the
+    /// burnchain?
+    pub fn is_inv_valid(&self, nk: &NeighborKey) -> bool {
+        !self.diverged_peers.contains(nk) && !self.broken_peers.contains(nk)
+    }
+
+    /// Is the inventory in this peer up-to-date with the given number of sortitions (even if it's diverged)?
+    pub fn is_inv_synced(&self, nk: &NeighborKey, min_num_sortitions: u64) -> bool {
+        if self.broken_peers.contains(nk) {
+            return false;
+        }
+
+        match self.block_stats.get(nk) {
+            None => false,
+            Some(ref stats) => stats.inv.num_sortitions >= min_num_sortitions
+        }
+    }
+
+    /// How many sortitions do we know about from this neighbor?
+    /// Ignores broken or diverged peers.
+    pub fn get_inv_sortitions(&self, nk: &NeighborKey) -> u64 {
+        if self.broken_peers.contains(nk) || self.diverged_peers.contains(nk) {
+            return 0;
+        }
+
+        match self.block_stats.get(nk) {
+            Some(stats) => stats.inv.num_sortitions,
+            _ => 0
+        }
+    }
+
+    /// Cull sync peers of broken peers
+    pub fn cull_broken_peers(&mut self) -> () {
+        for nk in self.broken_peers.iter() {
+            if self.sync_peers.contains(nk) {
+                test_debug!("Cull broken peer {:?}", nk);
+                self.sync_peers.remove(nk);
+            }
+        }
+    }
+
+    /// Cull sync peers of stale peers 
+    pub fn cull_stale_peers(&mut self) -> () {
+        for nk in self.stale_peers.iter() {
+            if self.sync_peers.contains(nk) {
+                test_debug!("Cull stale peer {:?}", nk);
+                self.sync_peers.remove(nk);
+            }
+        }
+    }
+
+    /// Cull sync peers of dead peers
+    pub fn cull_dead_peers(&mut self) -> () {
+        for nk in self.dead_peers.iter() {
+            if self.sync_peers.contains(nk) {
+                test_debug!("Cull dead peer {:?}", nk);
+                self.sync_peers.remove(nk);
+            }
+        }
+    }
+    
+    /// Cull sync peers of diverged peers
+    pub fn cull_diverged_peers(&mut self) -> () {
+        for nk in self.diverged_peers.iter() {
+            if self.sync_peers.contains(nk) {
+                test_debug!("Cull diverged peer {:?}", nk);
+                self.sync_peers.remove(nk);
+            }
+        }
+    }
+
+    /// Get the list of faulty peers
+    pub fn get_faulty_peers(&self) -> Vec<NeighborKey> {
+        let mut set = HashSet::new();
+        for nk in self.broken_peers.iter() {
+            set.insert(nk.clone());
+        }
+        for nk in self.dead_peers.iter() {
+            set.insert(nk.clone());
+        }
+
+        let mut ret = vec![];
```

Comment:
You can directly convert a HashSet into a Vec with `.into_iter().collect()`

Reply:
Fixed in f69fce59e